output/json/wrapper.go

```diff
@@ -26,38 +26,40 @@ import (
 	"github.com/loadimpact/k6/stats"
 )
 
+// Envelope is the data format we use to export both metrics and metric samples
+// to the JSON file.
 type Envelope struct {
 	Type   string      `json:"type"`
 	Data   interface{} `json:"data"`
 	Metric string      `json:"metric,omitempty"`
 }
 
-type JSONSample struct {
+// Sample is the data format for metric sample data in the JSON file.
+type Sample struct {
 	Time  time.Time         `json:"time"`
 	Value float64           `json:"value"`
 	Tags  *stats.SampleTags `json:"tags"`
 }
 
-func NewJSONSample(sample *stats.Sample) *JSONSample {
-	return &JSONSample{
+func newJSONSample(sample stats.Sample) Sample {
+	return Sample{
 		Time:  sample.Time,
 		Value: sample.Value,
 		Tags:  sample.Tags,
 	}
 }
 
-func WrapSample(sample *stats.Sample) *Envelope {
-	if sample == nil {
-		return nil
-	}
-	return &Envelope{
+// WrapSample is a used to package a metric sample in a way that's nice to
+// export to JSON.
+func WrapSample(sample stats.Sample) Envelope {
```

Comment:
Maybe unrelated to this PR, but it would be good to move this file to a common location for all outputs (`output/json.go` or `stats/json.go`?), since it seems generic enough, and it's already used by the Kafka output, so it's weird that it would depend on the JSON output package directly.

Reply:
Hmm I am not sure any other outputs would ever use the function, but we can easily move it to `/output` in the future, if that ever proves necessary. Or not, I don't really see the problem with the current explicit dependency on `output/json` from the `kafka` output. Isn't it actually a good thing, it doesn't cost us anything, but it somewhat shows the `kafka` output is encoding the metric samples it sends the same way as the `json` output?

Reply:
I slightly disagree, but it's not a big deal.